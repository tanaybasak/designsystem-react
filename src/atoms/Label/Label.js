import React from 'react';
import PropTypes from 'prop-types';
import prefix from '../../settings';

const Label = ({ className, children, ...restProps }) => {
    const classnames = `${prefix}-label ${className}`.trim();

    return (
        <label className={classnames} {...restProps}>
            {children}
        </label>
    );
}

Label.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    htmlFor: PropTypes.string,
    onClick: PropTypes.func
};

Label.defaultProps = {
    children: null,
    className: "",
    htmlFor: "",
    onClick: () => { }
};

export default Label;
