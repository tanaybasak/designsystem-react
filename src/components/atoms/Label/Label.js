import React from 'react';
import PropTypes from 'prop-types';
import { prefix } from '../../../settings';

export default function Label({ className, children, ...labelProps }) {

    return (
        <label
            className={`${prefix}-label ${className}`}
            {...labelProps}
        >
            <span>{children}</span>
        </label>
    )
};

Label.propTypes = {
    children: PropTypes.node.isRequired,
    className: PropTypes.string,
    htmlFor: PropTypes.string,
    onClick: PropTypes.func
};

Label.defaultProps = {
    children: '',
    className: '',
    htmlFor: '',
    onClick: () => { }
};