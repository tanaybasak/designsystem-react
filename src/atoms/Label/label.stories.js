import React from 'react';
import { storiesOf } from '@storybook/react';
//@update-path-build-start
import Label from './Label';
//@update-path-build-end

storiesOf('Label', module).add('default', () => <Label>This is label</Label>, {
  info: {
    text: `Description About Label Component\n
      import { Label } from '@patron/patron-react/label';`
  }
});
