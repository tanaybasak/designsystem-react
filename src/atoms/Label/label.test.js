import React from 'react';
import renderer from 'react-test-renderer';

import Label from './Label';

it('Label renders correctly', () => {
  const tree = renderer.create(<Label />).toJSON();
  expect(tree).toMatchSnapshot();
});
