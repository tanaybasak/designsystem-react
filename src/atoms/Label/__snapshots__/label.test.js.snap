// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<label
  className="hcl-label"
  htmlFor=""
/>
`;
